Automattic/jetpack
# Refresh the site's modules and settings after successful product activation

Fixes 1201096622142517-as-1201431693192886

#### Changes proposed in this Pull Request:

* Adds a call to `fetchModules` and `fetchSettings` after successful activation of a product. We need these calls to provide a consistent UX that is aligned with the user's expectations. For example, after a user purchases Jetpack Complete, they will expect to have the search feature enabled and working. Without these calls, the user would have to refresh the page to see that. This PR prevents that.

**Note**: there is a [known issue that affects the activation of Jetpack Complete](1201096622142517-as-1201400491645539). Particularly, the request to activate the license times out. You can bypass that by increasing the timeout from 10 to 20-30 seconds in the `projects/packages/connection/src/class-client.php` file (line 82).

#### Jetpack product discussion


* Project Thread: p1HpG7-cSg-p2

#### Does this pull request change what data or activity we track or use?


* Not at all.

#### Testing instructions:






* Spin up a site using this Jetpack branch.
* Connect Jetpack to your A8C WordPress.com account.
* Enable the A8c Proxy.
* Go to https://cloud.jetpack.com/pricing.
* Select Jetpack Complete.
* Complete checkout.
* On the "Thank you for purchase!" page, just stop there.. (Don't enter a site URL in the input box).
* On your site, visit `/wp-admin/admin.php?page=jetpack#/license/activation`.
* Enter the license key associated with your Jetpack Complete purchase (you can get it from SA).
* Activate the product.
* Once the activation finishes, do not refresh the page. 
* Click on Jetpack > Dashboard (sidebar).
* Click on the My Plan tab.
* Verify that you see Jetpack Complete in the My Plan section.
* Verify that you see `Instant search and filtering` card just below that.
* Go to Jetpack > Settings (sidebar).
* Click on the Performace tab.
* Verify that Search and Instant Search Experience are enabled.

https://user-images.githubusercontent.com/3418513/143627729-b2993f6b-3214-4c17-99c6-82fd3d9a134e.mp4
